lib/questions.rb

```diff
@@ -63,33 +99,51 @@ def separate_array_into_even_and_odd_numbers(array)
 # e.g. 'bob'. So in the array ['bob', 'radar', 'eat'], there
 # are 2 palindromes (bob and radar), so the method should return 2
 def number_of_elements_that_are_palindromes(array)
+  count = 0
+  array.each do |element|
+    if element == element.reverse then
+      count += 1
+    end
+  end
+  count
 end
 
 # return the shortest word in an array
 def shortest_word_in_array(array)
+  shortest = array.sort { |x, y| x.size <=> y.size }
+  shortest[0]
 end
 
 # return the shortest word in an array
 def longest_word_in_array(array)
+  longest = array.sort { |x, y| x.size <=> y.size }
+  longest[-1]
 end
 
 # add up all the numbers in an array, so [1, 3, 5, 6]
 # returns 15
 def total_of_array(array)
+  array.inject(0){|result, element| result += element}
 end
 
 # turn an array into itself repeated twice. So [1, 2, 3]
 # becomes [1, 2, 3, 1, 2, 3]
 def double_array(array)
+  arrayCopy = array.clone
+  arrayCopy.map {|element| array << element}
+  array
 end
 
 # convert a symbol into a string
 def turn_symbol_into_string(symbol)
+  symbol.to_s
 end
 
 # get the average from an array, rounded to the nearest integer
 # so [10, 15, 25] should return 17
 def average_of_array(array)
+  total = array.inject {|total, element| total += element}
```

Comment:
Shadowing outer local variable - `total`.<br>Space between { and | missing.<br>Space missing inside }.